posthog/api/feature_flag.py

```diff
@@ -1578,6 +1621,23 @@ def handle_feature_flag_change(sender, scope, before_update, after_update, activ
         ),
     )
 
+    # Invalidate flag definitions cache when feature flags change
+    invalidate_cache_for_feature_flag_change(after_update, activity)
+
+
+@receiver(post_save, sender=Cohort)
+@receiver(post_delete, sender=Cohort)
+def handle_cohort_change(sender, instance, **kwargs):
```

Comment:
Does cover the case when a cohort is recalcualted?

Reply:
Good question. I assume that when we recalculate a cohort, we save it. But I'll test it.

Reply:
So I confirmed that recalculating a cohort will invalidate the cache. I think for now, that's fine. We can tighten up the cache invalidation logic later.